From 25ed4bd51b05d66bf0014a98d260734849e4a973 Mon Sep 17 00:00:00 2001
From: Michael Weiss <dev.primeos@gmail.com>
Date: Sat, 27 Apr 2019 14:26:16 +0200
Subject: [PATCH] Load configs from /etc but fallback to /nix/store

This change will load all configuration files from /etc, to make it easy
to override them, but fallback to /nix/store/.../etc/sway/config to make
Sway work out-of-the-box with the default configuration on non NixOS
systems.

Signed-off-by: Martin Schwaighofer <mschwaig@users.noreply.github.com>
---
 meson.build   | 3 ++-
 sway/config.c | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index b845d437..3e5d7961 100644
--- a/meson.build
+++ b/meson.build
@@ -164,7 +164,8 @@ if scdoc.found()
 	endforeach
 endif
 
-add_project_arguments('-DSYSCONFDIR="/@0@"'.format(join_paths(prefix, sysconfdir)), language : 'c')
+add_project_arguments('-DSYSCONFDIR="/@0@"'.format(sysconfdir), language : 'c')
+add_project_arguments('-DNIX_SYSCONFDIR="/@0@"'.format(join_paths(prefix, sysconfdir)), language : 'c')
 
 version = '"@0@"'.format(meson.project_version())
 git = find_program('git', native: true, required: false)
diff --git a/sway/config.c b/sway/config.c
index 76b9ec08..ee8abaed 100644
--- a/sway/config.c
+++ b/sway/config.c
@@ -374,7 +374,8 @@ static char *get_config_path(void) {
 		{ .prefix = home, .config_folder = ".i3"},
 		{ .prefix = config_home, .config_folder = "i3"},
 		{ .prefix = SYSCONFDIR, .config_folder = "sway"},
-		{ .prefix = SYSCONFDIR, .config_folder = "i3"}
+		{ .prefix = SYSCONFDIR, .config_folder = "i3"},
+		{ .prefix = NIX_SYSCONFDIR, .config_folder = "sway/config"}
 	};
 
 	size_t num_config_paths = sizeof(config_paths)/sizeof(config_paths[0]);
-- 
2.30.1

